homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py

```diff
@@ -0,0 +1,134 @@
+"""Support for Atlantic Pass APC Heating And Cooling Zone Control."""
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+from pyoverkiz.models import Command
+
+from homeassistant.components.climate import (
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+PRESET_FROST_PROTECTION = "frost_protection"
+
+OVERKIZ_TO_HVAC_MODE: dict[str, str] = {
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.ECO: HVACMode.AUTO,
+    OverkizCommandParam.MANU: HVACMode.HEAT,
+    OverkizCommandParam.HEATING: HVACMode.HEAT,
+    OverkizCommandParam.STOP: HVACMode.OFF,
+    OverkizCommandParam.INTERNAL_SCHEDULING: HVACMode.AUTO,
+    OverkizCommandParam.COMFORT: HVACMode.HEAT,
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+
+OVERKIZ_TO_PRESET_MODES: dict[str, str] = {
+    OverkizCommandParam.OFF: PRESET_ECO,
+    OverkizCommandParam.STOP: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.MANU: PRESET_COMFORT,
+    OverkizCommandParam.ABSENCE: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.INTERNAL_SCHEDULING: PRESET_COMFORT,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+THERMAL_CONFIGURATION_STATE = "core:ThermalConfigurationState"
+
+
+class AtlanticPassAPCHeatingAndCoolingZone(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Zone Control."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODE[
+            cast(str, self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_MODE))
+        ]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        commands = [
+            Command(
+                OverkizCommand.SET_PASS_APC_HEATING_MODE,
+                [HVAC_MODE_TO_OVERKIZ[hvac_mode]],
+            ),
+            # We also needs to execute these 3 commands to make it work correctly
+            Command(
+                OverkizCommand.SET_DEROGATION_ON_OFF_STATE,
+                [OverkizCommandParam.OFF],
+            ),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_MODE),
+            Command(OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE),
```

Comment:
I am wondering if we should wrap it with `Command()` in the python-overkiz-api, so that we don't need to do it here. For the `async_execute_command` we allow passing a string, so that you don't need to wrap this with `Command()` every time.

Reply:
Indeed, it would reduce the verbosity, and now we have also OverkizCommand we can use instead of the string.

Reply:
So you're saying I should update the new `async_execute_commands` to receive an Array like:  
```python
commands = [
  [OverkizCommand.SET_PASS_APC_HEATING_MODE, [HVAC_MODE_TO_OVERKIZ[hvac_mode]]],
  [OverkizCommand.SET_DEROGATION_ON_OFF_STATE, [OverkizCommandParam.OFF]],
  OverkizCommand.REFRESH_PASS_APC_HEATING_MODE,
  OverkizCommand.REFRESH_PASS_APC_HEATING_PROFILE
]
```

And then instantiate the Command inside the `async_execute_commands`.

Is that what you're asking?

I'm not a Python dev at all, so I might code it the junior way.

Reply:
I updated my PR.
Please let me know if it's what you're excepting for.  
If it's ok, I'll update my other PR #78665 as well.

Reply:
@nyroDev We are talking about this [method](https://github.com/iMicknl/python-overkiz-api/blob/main/pyoverkiz/client.py#L622) within the overkiz client API.

Reply:
@tetienne that's the method I'm using in the new `async_execute_commands` I wrote in the executor here.
This new function do the same as `async_execute_command`: expect commands as strings (or list if parameters are needed), wrap it into `Command` and call the `execute_commands` function API.
I could do a PR on the API to do what you're saying, but I guess it will be a breaking change, requesting more changes in many other projects I think.

Reply:
Closing this thread in favor of splitting to another PR #79576